ha_mqtt_discoverable/__init__.py

```diff
@@ -576,62 +578,89 @@ def __init__(self, settings: Settings[EntityType]) -> None:
         self._settings = settings
         self._entity = settings.entity
 
-        self.topic_prefix = (
-            f"{self._settings.mqtt.topic_prefix}/"
-            f"{self._entity.component}/{clean_string(self._entity.name)}"
+        # TODO: use different state topic than homeassistant. Maybe start with `hmd/`?
```

Comment:
If we use a different prefix, won't that make HA not autodiscover our entities?

Maybe have a `discovery_topic_prefix` argument that defaults to `homeassistant`?

Reply:
I was thinking to avoid "invading" the `homeassistant/` topic tree with our state messages, to keep some sorts of "boundaries" between components.
For instance whoever subscribes to `homeassistant/#` I suppose is interested in messages regarding HA and strictly about HA, in that case it does not care about receiving our sensors update states.
Changing the prefix should not be an issue, since as part of the `config` message we send to HA there is the `state_topic` attribute, so HA knows the topic to listen to. It is the `config` message that must be [sent specifically to HA](https://www.home-assistant.io/integrations/mqtt/#discovery-topic), inside the `homeassistant/` prefix.
I am thinking for instance about shelly devices, where they all operate under the [`shellies/` topic](https://shelly-api-docs.shelly.cloud/gen1/#availability-and-announces), and I have them integrated without problems in my using [MQTT discovery](https://github.com/bieniu/ha-shellies-discovery).

Reply:
Good point. `hmd` seems like a better default prefix for the status messages then. Using our own prefix will be cleaner and as a side benefit make it easier to see at a glance how many `hmd`-created entities are being published to the `mqtt` server.

Reply:
Yeah, that the advantage of havinga a dedicated topic! 😉